transactions/pendingtxtracker_test.go

```diff
@@ -265,17 +265,29 @@ func TestPendingTxTracker_MultipleClients(t *testing.T) {
 		(*res)["blockNumber"] = transactionSuccessStatus
 	})
 
-	for i := range txs {
-		err := m.TrackPendingTransaction(txs[i].ChainID, txs[i].Hash, txs[i].From, txs[i].Type, true)
-		require.NoError(t, err)
-	}
+	// If we call TrackPendingTransaction immediately, there is a chance that some events
+	// will be emitted before we reach select, so occasionally select fails on timeout.
+	go func() {
```

Comment:
Starting to do it async is not a good idea for a test. We miss the control flow and increase the chance of mixing the update with `EventPendingTransactionUpdate` with `EventPendingTransactionStatusChanged`.